electrum/lnworker.py

```diff
@@ -1563,18 +1564,19 @@ def create_routes_for_payment(
                                 self.logger.info('not enough margin to pay trampoline fee')
                                 raise NoPathFound()
                     else:
-                        for (chan_id, _), part_amount_msat in s[0].items():
-                            if part_amount_msat:
-                                channel = self.channels[chan_id]
-                                route = self.create_route_for_payment(
-                                    amount_msat=part_amount_msat,
-                                    invoice_pubkey=invoice_pubkey,
-                                    min_cltv_expiry=min_cltv_expiry,
-                                    r_tags=r_tags,
-                                    invoice_features=invoice_features,
-                                    channels=[channel],
-                                    full_path=None)
-                                routes.append((route, part_amount_msat, final_total_msat, part_amount_msat, min_cltv_expiry, payment_secret, fwd_trampoline_onion))
+                        for (chan_id, _), part_amounts_msat in s[0].items():
+                            if part_amounts_msat:
+                                for part_amount_msat in part_amounts_msat:
```

Comment:
isn't the if statement redundant ?

Reply:
It's because the returned dict is a defaultdict that can have empty lists, see:
https://github.com/spesmilo/electrum/blob/49470a34f63b2bd68256bb671471882ef3257fe1/electrum/tests/test_mpp_split.py#L29-L38
I will remove those empty entries.

Reply:
Yes, it was redundant :)